### PR TITLE
Add RAGE engine

### PR DESCRIPTION
**Pull request by Lifeismana**
### SteamDB app page links to a few games using this
https://steamdb.info/app/1174180/
https://steamdb.info/app/271590/
https://steamdb.info/app/12210/
### Brief explanation of the change
Added the Rage Engine

### CODE REVIEW

File: descriptions/Engine.RAGE.md [hunk 1]
```diff
@@ -0,0 +1 @@
+[**RAGE (Rockstar Advanced Game Engine)**] is a proprietary game engine developed by RAGE Technology Group, a division of Rockstar Games. It was first used in the game Rockstar Games Presents Table Tennis released in 2006. Since then, Rockstar has kept developing it.
```
Comment (larsiusprime) [inline descriptions/Engine.RAGE.md:1]: Should include a note that RAGE is not to be confused with id Tech 5, which was first used in the id Software title "Rage", and is sometimes colloquially called "The Rage Engine"

https://rage.fandom.com/wiki/Id_Tech_5

Comment (Lifeismana) [reply descriptions/Engine.RAGE.md]: Done

**Comment by larsiusprime** [comment]
Here's all the depots that have rpf files, can you take a peek and check for false positives?

* [12211 - Grand Theft Auto IV Data](https://steamdb.info/depot/12211/)
* [12212 - Grand Theft Auto IV Data A](https://steamdb.info/depot/12212/)
* [12221 - Grand Theft Auto IV Episodes from Liberty City_Data](https://steamdb.info/depot/12221/)
* [204104 - MP3_Data_Final](https://steamdb.info/depot/204104/)
* [204112 - DisorganizedCrime](https://steamdb.info/depot/204112/)
* [204114 - OSX Depot](https://steamdb.info/depot/204114/)
* [211520 - LocalJustice](https://steamdb.info/depot/211520/)
* [211521 - HostageNegotiations](https://steamdb.info/depot/211521/)
* [211522 - PainfulMemories](https://steamdb.info/depot/211522/)
* [211523 - DeathmatchMadeInHeaven](https://steamdb.info/depot/211523/)
* [211524 - OSX DLC - All Preorder](https://steamdb.info/depot/211524/)
* [211525 - OSX DLC4 - Hostage Negotiations](https://steamdb.info/depot/211525/)
* [211526 - OSX DLC5 - Painful Memories](https://steamdb.info/depot/211526/)
* [211527 - OSX DLC6 - Deathmatch Made in Heaven](https://steamdb.info/depot/211527/)
* [211529 - OSX DLC2 - LocalJustice](https://steamdb.info/depot/211529/)
* [224901 - Iron Sky Content PC](https://steamdb.info/depot/224901/)
* [224904 - Iron Sky Content Linux](https://steamdb.info/depot/224904/)
* [227961 - Iron Sky Invasion Demo Content](https://steamdb.info/depot/227961/)
* [233211 - Air Conflicts: Vietnam Content](https://steamdb.info/depot/233211/)
* [233220 - Air Conflicts: Vietnam Linux](https://steamdb.info/depot/233220/)
* [233221 - Air Conflicts: Vietnam OSX](https://steamdb.info/depot/233221/)
* [271593 - Shetland Pony Depot](https://steamdb.info/depot/271593/)
* [271594 - Shetland Pony Depot](https://steamdb.info/depot/271594/)
* [271595 - Shetland Pony Depot](https://steamdb.info/depot/271595/)
* [493451 - Wild Romance Content](https://steamdb.info/depot/493451/)
* [493452 - Wild Romance JP](https://steamdb.info/depot/493452/)
* [494161 - Lion Heart Content](https://steamdb.info/depot/494161/)
* [545342 - Companion Traditional Chinese](https://steamdb.info/depot/545342/)
* [572142 - Koi Musubi English Demo Depot](https://steamdb.info/depot/572142/)
* [572143 - Koi Musubi TW Demo Depot](https://steamdb.info/depot/572143/)
* [572144 - Koi Musubi CN Demo Depot](https://steamdb.info/depot/572144/)
* [629962 - 《夏花的轨迹——A summer promise to forever》 Traditional Chinese](https://steamdb.info/depot/629962/)
* [863871 - Brain 43℃ Content](https://steamdb.info/depot/863871/)
* [863872 - Brain 43℃cn](https://steamdb.info/depot/863872/)
* [863873 - Brain 43℃tc](https://steamdb.info/depot/863873/)
* [863874 - Brain 43℃ru](https://steamdb.info/depot/863874/)
* [869871 - She is mermaid - base - SC](https://steamdb.info/depot/869871/)
* [873291 - Depot Common](https://steamdb.info/depot/873291/)
* [882941 - The Moment We Met Content](https://steamdb.info/depot/882941/)
* [1033651 - 寄给明日的希望 Content](https://steamdb.info/depot/1033651/)
* [1174182 - Red Dead Redemption 2 Base](https://steamdb.info/depot/1174182/)
* [1174184 - Red Dead Redemption 2 Update](https://steamdb.info/depot/1174184/)
* [1174185 - Red Dead Redemption 2 Packs](https://steamdb.info/depot/1174185/)
* [1174186 - Red Dead Redemption 2 Audio](https://steamdb.info/depot/1174186/)
* [1338461 - Jiangshi x Daoshi Content](https://steamdb.info/depot/1338461/)
* [1524770 - 戾谈 红色录像带 绯红 (1524770) 个 Depot](https://steamdb.info/depot/1524770/)

**Comment by Lifeismana** [comment]
Wow quite a lot of false positive in there

**Comment by Lifeismana** [comment]
Found the solution!
I need to do a 2 pass on this one to search for `.rpf` files and a `metadata.dat`.
Should i put this check at the top or at the bottom of TryDeduceEngine ?

**Comment by larsiusprime** [comment]
Bottom is generally nice for things that you expect a lot of false positives on because that way you rule a bunch of other stuff out first

**Comment by Lifeismana** [comment]
Then i'll put it at the top because checking for a `metadata.dat` removes every false positive

**Pull request merged**